mypy/semanal.py

```diff
@@ -1063,16 +1054,71 @@ def process_typevar_declaration(self, s: AssignmentStmt) -> None:
             else:
                 self.fail("Cannot redefine '%s' as a type variable" % name, s)
             return
-        if len(call.args) > 1:
+
+        # Contraining types
+        n_values = call.arg_kinds[1:].count(ARG_POS)
+        if n_values > 0:
             # Analyze enumeration of type variable values.
-            values = self.analyze_types(call.args[1:])
+            values = self.analyze_types(call.args[1:1+n_values])
         else:
             # Type variables can refer to an arbitrary type.
             values = []
+
+        # Parameters
```

Comment:
This method is getting pretty big. I suggest refactoring into a few smaller methods.